### DISCUSSION TITLE
Artificially update variables after each time step in TimeDerivative kernel

### DISCUSSION THREAD

**Discussion by Joseph-0123** [idaholab/moose#23158] [Q&A General] [answered]
Dear all,

I have a question.  After each time step, we solve the PDEs and get the variable values. 

But now if I want to artificially do some changes to one variable, for example, multiply it with a constant value after each time step and use the artificially changed value as the input (or old value) for the next time step. 

How to easily achieve that?

Thanks in advance.

J

**Comment by GiudGiud** [comment, marked as answer]
Hello

So we generally dont allow that, because a nonlinear variable is the output of solver an equation not any other pre-determined operation like multiplying it. If you just want a multiplied value from a variable, that's more what the auxiliary variables are for. 

However, if you do want to do that, you can use a MultiApp to do that:
- transfer the non linear variable to an auxvariable in a MultiApp on timestep_end
- use an auxkernel to multiply the variable there, for example the ParsedAux
- transfer the value back from the auxvariable to the nonlinear variable in the main app


Another option is to write a custom user object to do the multipliciation. This requires coding.

Guillaume

**Reply by Joseph-0123** [reply]
> Hello
> 
> So we generally dont allow that, because a nonlinear variable is the output of solver an equation not any other pre-determined operation like multiplying it. If you just want a multiplied value from a variable, that's more what the auxiliary variables are for.
> 
> However, if you do want to do that, you can use a MultiApp to do that:
> 
> * transfer the non linear variable to an auxvariable in a MultiApp on timestep_end
> * use an auxkernel to multiply the variable there, for example the ParsedAux
> * transfer the value back from the auxvariable to the nonlinear variable in the main app
> 
> Another option is to write a custom user object to do the multipliciation. This requires coding.
> 
> Guillaume

Thanks for your suggestions！I will try all of them.
